QuizWhois.Domain/Services/Implementations/QuizService.cs

```diff
@@ -57,6 +59,37 @@ public async Task AddToQuiz(AddToSetModel addToSetModel)
             _logger.LogInformation(messageToLog);
         }
 
+        public QuizModel GetQuiz(long quizId)
+        {
+            if (quizId <= 0)
+            {
+                throw new Exception("Id was invalid number");
+            }
+
+            var entity = _dbContext.Quizzes.Select(x => new
+            {
+                Id = x.Id,
+                Name = x.Name,
+                Questions = x.Questions.Select(y => new Question
+                {
+                    Id = y.Id,
+                    QuizId = y.QuizId,
+                    QuestionText = y.QuestionText,
+                    CorrectAnswers = y.CorrectAnswers.ToList()
+                }).ToList()
+            }).FirstOrDefault(x => x.Id == quizId);
```

Comment:
I know lazy load is doing it's magic, but it is better imo to find by id first, then select fields of resulting query. It should result in less cpu work since call to Select is evaluated at client.

Reply:
As far i understand, it's not lazy, it's eager loading, but i noticed, that i have to filter at first, ty, fixed.